recipe/meta.yaml

```diff
@@ -132,12 +131,15 @@ outputs:
   - name: root-dependencies
     build:
       number: {{ build_number }}
+      string: py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ build_number }}
       skip: True  # [win]
       script: echo "Minimal build script is required for conda to set RECIPE_DIR"
       run_exports:
         # https://abi-laboratory.pro/?view=timeline&l=root
         - {{ pin_subpackage('root-dependencies', max_pin='x.x.x') }}
     requirements:
+      # Include the build dependencies to ensure the build string is correct
+      build: {{- build_dependencies()|replace("\n  ", "\n        ") }}
```

Comment:
Do we need the string and this?

Reply:
Yes as it doesn’t depend gfortran without it.

I’m not sure though and it seems like there is a bug in conda build. I think the jinja2 isn’t being fully expanded before the hash is computed.

Reply:
Could we add just the Python and/or fortran part, here? It wouldn't be too bad to list those each time. (And then file an issue, probably, if that fixes it)

Reply:
It's actually worse as it was also breaking the run exports. I've changed it to YAML anchors for now but it's limited as we can't add anything to the list. I'll try to make a minimal reproducer and submit a bug report.